org.sofproject.alsa.topo/src/org/sofproject/alsa/topo/model/AlsaTopoGraph.java

```diff
@@ -536,7 +515,35 @@ public void serialize() throws CoreException, IOException {
 		}
 		outputFile.getParent().refreshLocal(1, null);
 	}
-
+	
+	@Override
+	public String getPipelineString() {
+		try {
+			ByteArrayOutputStream os = new ByteArrayOutputStream();
```

Comment:
If you need a string, you may use the StringBuffer, not a byte representation that might be converted into characters and back.